scripts/combine-profile-information.py

```diff
@@ -0,0 +1,127 @@
+
+args = parser.parse_args()
+
+DUMP_DIR = Path(args.dump_dir)
+if not DUMP_DIR.exists():
+    raise ValueError(f"{DUMP_DIR} does not exist")
+
+if args.summary_path is None:
+    args.summary_path = DUMP_DIR.parent / "summary.csv"
+SUMMARY_PATH = Path(args.summary_path)
+
+CsvRow = Dict[str, str]
+CsvRows = List[CsvRow]
+SummaryRow = Dict[str, str]
+
+
+def parse_counters(counters: str) -> Dict[str, str]:
+    if not counters:
+        return {}
+    split_counters = counters.strip("[]").split("; ")
+    return dict(map(lambda x: tuple(x.split(" = ")), split_counters))
+
+
+def get_csv_rows(csv_path: Path) -> CsvRows:
+    with open(csv_path) as csv_file:
+        csv_reader = csv.DictReader(csv_file)
+        return list(csv_reader)
+
+
+def get_input_csv_paths() -> Iterator[Tuple[Path, CsvRows]]:
+    return (csv_path for csv_path in DUMP_DIR.glob("*.csv"))
+
+
+def split_into_number_and_unit(number_string: str) -> (Union[int, float], str):
+    numeric = "".join(map(str, range(10))) + "."
+    unit_start_position = next(
+        (i for i, c in enumerate(number_string) if c not in numeric), len(number_string)
+    )
+    number = float(number_string[:unit_start_position])
+    unit = number_string[unit_start_position:]
+    if number.is_integer():
+        number = int(number)
+    return number, unit
+
+
+PRIMARY_KEY = "pass name"
+summary_non_counter_fields = set()
+summary_counter_fields = set()
+
+for rows in map(get_csv_rows, get_input_csv_paths()):
+    curr_fields = set(rows[0])
+    curr_fields.remove(PRIMARY_KEY)
+    if "counters" in curr_fields:
+        curr_fields.remove("counters")
+        for row in rows:
+            summary_counter_fields.update(parse_counters(row["counters"]))
+    summary_non_counter_fields.update(curr_fields)
+
+# CR mitom: Not supporting memory for now
+summary_non_counter_fields -= {
+    "alloc",
+    "top-heap",
+    "absolute-top-heap",
+}
+
+# Ensure consistent ordering of summary fields
+field_collections = {PRIMARY_KEY}, summary_non_counter_fields, summary_counter_fields
+SUMMARY_FIELD_NAMES = sum(map(lambda fields: sorted(fields), field_collections), [])
+
+units = {}
+for field in summary_non_counter_fields:
+    first_row = get_csv_rows(next(get_input_csv_paths()))[0]
+    _, units[field] = split_into_number_and_unit(first_row[field])
+
+
+def strip_units(row: CsvRow) -> CsvRow:
+    return {k: (v[: -len(units[k])] if k in units else v) for k, v in row.items()}
+
+
+def row_summary(row: CsvRow) -> SummaryRow:
+    full_summary = {**strip_units(row), **parse_counters(row["counters"])}
+    return {k: v for k, v in full_summary.items() if k in SUMMARY_FIELD_NAMES}
+
+
+def csv_to_summaries(rows: CsvRows) -> Iterable[SummaryRow]:
+    counter_rows = [row for row in rows if row["counters"]]
+    return map(row_summary, counter_rows)
+
+
+def split_into_number_and_unit(number_string: str) -> (str, Union[int, float]):
+    number_chars = "".join(str(i) for i in range(10)) + "."
```

Comment:
```suggestion
    number_chars = string.digits + "."
```
(Also a duplicate, that could be factored out.)